js/utils/metrics.js

```diff
@@ -152,3 +162,49 @@ export function endAjaxEvent(key, segmentation) {
       }]);
   }
 }
+
+/** Use when a component needs to pass a different prefix for a recorded event, usually passed in by
+ * the parent. If no prefix is passed in, the event will be named with just the eventName.
+ * @param opts.prefix(string)
+ * @param opts.eventName(string)
+ * @paren opts.segmentation(object)
+ */
+export function recordPrefixedEvent(opts) {
+  if (!opts.eventName) throw new Error('Please provide an eventName');
```

Comment:
Generally, you want to explicitly declare required arguments that don't have any default, with the options hash at the end, for example:

`recordPrefixedEvent(eventName, segmentation, opts = {})`

I suppose a possible exception to that is if you have a whole lot of required ones, but even then you'd probably want to semantically seperate required and options ones, e.g:

`recordPrefixedEvent(manyManyReq = {}, opts = {})`

(required args are fine within the opts hash, but if they are given defaults. It's a bit odd, IMHO to put ones that the user is forced to provide in the options hash)